bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -450,42 +441,33 @@ public int available() {
   public ResultScanner<FlatRow> readFlatRows(ReadRowsRequest request) {
     // Delegate all resumable operations to the scanner. It will request a non-resumable scanner
     // during operation.
-    // TODO(sduskis): Figure out a way to perform operation level metrics with the
-    // AbstractBigtableResultScanner implementations.
-    return new ResumingStreamingResultScanner(retryOptions, request, streamingScannerFactory,
-        readRowsAsync.getRpcMetrics());
-  }
-
-  private ResultScanner<FlatRow> streamRows(ReadRowsRequest request) {
-    final Timer.Context timerContext = readRowsAsync.getRpcMetrics().timeRpc();
-    final AtomicBoolean wasCanceled = new AtomicBoolean(false);
-
-    CallOptions callOptions = getCallOptions(readRowsAsync.getMethodDescriptor(), request);
-    final ClientCall<ReadRowsRequest, ReadRowsResponse> readRowsCall =
-        readRowsAsync.newCall(callOptions);
-
-    ResponseQueueReader reader = new ResponseQueueReader(
-        retryOptions.getReadPartialRowTimeoutMillis(), retryOptions.getStreamingBufferSize());
-
-    final StreamObserverAdapter<ReadRowsResponse> listener =
-        new StreamObserverAdapter<>(readRowsCall, new RowMerger(reader));
+    ResponseQueueReader reader = new ResponseQueueReader(retryOptions.getStreamingBufferSize());
+    final ReadRowsRetryListener listener = createReadRowsRetryListener(request, reader);
+    return new ResumingStreamingResultScanner(reader, new ScanHandler() {
```

Comment:
Might be cleaner to have the listener implement ScanHandler. I know that you had it that way in the previous iteration, sorry to flip flop on this.